### PR TITLE
BTFSBTT-189: Display metadata nicely and BTFSBTT-209

### PR DESCRIPTION
**Pull request by steveyeom**
BTFSBTT-189/209 code changes:

Local unit tests successful exception one authentication error. 
This happens without my PR. So my local machine issue. 

--- FAIL: TestGoGetSupport (0.01s)
    gateway_test.go:565: test server url: http://127.0.0.1:39953
    gateway_test.go:580: status is 404, expected 200

### CODE REVIEW

File: core/commands/object/object.go [hunk 1]
```diff
@@ -24,6 +25,12 @@ type Node struct {
 	Data  string
 }
 
+type NodeWithMeta struct {
+	Links    []Link
+	Data     string
```
Comment (taiyangc) [inline core/commands/object/object.go]: You can put it like this

```go
type NodeWithMeta struct {
    Node

    Metadata string
}
```

Comment (taiyangc) [reply core/commands/object/object.go]: Should `Metadata` be a `map[string]interface{}`? We talked about unmarshalling it for better display (no quotes)?

File: core/commands/object/object.go [hunk 2]
```diff
@@ -86,15 +94,20 @@ is the raw data of the object.
 	Arguments: []cmds.Argument{
 		cmds.StringArg("key", true, false, "Key of the object to retrieve, in base58-encoded multihash format.").EnableStdin(),
 	},
+	Options: []cmds.Option{
+		cmds.BoolOption(metaDisplayOptionName, "m", "Write token metadata"),
```
Comment (taiyangc) [inline core/commands/object/object.go]: "Display token metadata."

File: core/commands/object/object.go [hunk 3]
```diff
@@ -584,6 +589,63 @@ func getObject(res cmds.ResponseEmitter, r io.Reader, dfenc string, nd ipld.Node
...
+
+	for i, link := range nd.Links() {
+		node.N.Links[i] = Link{
+			Hash: enc.Encode(link.Cid),
+			Name: link.Name,
+			Size: link.Size,
+		}
+	}
+
+	return *node, nil
```
Comment (taiyangc) [inline core/commands/object/object.go]: Should return a pointer type.. the previous commit was using `*NodeWithMeta`

Comment (taiyangc) [reply core/commands/object/object.go]: Pointer is more efficient.

**Comment by steveyeom** [comment]
Richard is ok to merge this PR. Local unit tests under Docker run successfully.

**Comment by taiyangc** [comment]
What is the current issue?

**Comment by steveyeom** [comment]
Currently the jjenkins unit tests get aborted for any PR since
readonly_unit are stuck. Richard is working on that with others.
That is why my PR got aborted. But I have merged as Richard is ok with that
given that my local runs works find and the changes after
the last successful run are all syntax related and does not affect existing
unit tests.

**Pull request merged**